src/diem/testing/miniwallet/app/app.py

```diff
@@ -244,3 +300,21 @@ def _gen_subaddress(self, account_id: str) -> bytes:
         sub = self.store.next_id().to_bytes(8, byteorder="big")
         self.store.create(Subaddress, account_id=account_id, subaddress_hex=sub.hex())
         return sub
+
+    def validate_unique_reference_id(self, reference_id: str) -> None:
+        try:
+            duplicate_ref_id = self.store.find(ReferenceID, reference_id=reference_id)
+            raise ValueError(f"{reference_id} exists for account ID {duplicate_ref_id.account_id}")
+        except NotFoundError:
+            return
+
+    def _validate_unique_reference_id(self, reference_id: Dict[str, Any]) -> None:
```

Comment:
param name `reference_id` -> `data`, but it probably better to inline this method: `lambda _: self.validate_unique_reference_id(reference_id)`